paddle/fluid/operators/lookup_table_op.cc

```diff
@@ -131,7 +155,7 @@ class LookupTableOpGrad : public framework::OperatorWithKernel {
  protected:
   framework::OpKernelType GetExpectedKernelType(
       const framework::ExecutionContext& ctx) const override {
-    auto data_type = framework::GetDataTypeOfVar(ctx.InputVar("Out"));
+    auto data_type = framework::GetDataTypeOfVar(ctx.InputVar("W"));
```

Comment:
Please use `framework::GradVarName("Out")` here, since `W` is not always initialized especially in distributed mode.

Reply:
Done.